src/machine-learning/machine-learning.ts

```diff
@@ -279,18 +286,55 @@ export class Model {
     }
   }
 
+  /**
+   * Locked indicates if there are active long running operations on the model.
+   * Models may not be modified when they are locked.
+   */
   public get locked(): boolean {
     // Backend does not currently return locked models.
     // This will likely change in future.
-    return false;
+    return (this.activeOperations?.length || 0) > 0;
   }
 
-  /* eslint-disable-next-line @typescript-eslint/no-unused-vars */
-  public waitForUnlocked(maxTimeSeconds?: number): Promise<void> {
+  /**
+   * Wait for the active operations on the model to complete.
+   * @param maxTimeSeconds The number of seconds to wait for the model to be unlocked. 0 for default.
+   */
+  public waitForUnlocked(maxTimeSeconds = 0): Promise<void> {
     // Backend does not currently return locked models.
     // This will likely change in future.
+    if ((this.activeOperations?.length || 0) > 0) {
+      const client = new MachineLearningApiClient(this.app!);
+      return client.handleOperation(this.activeOperations![0], true, maxTimeSeconds)
+        .then((modelResponse) => this.updateFromResponse(modelResponse));
+    }
     return Promise.resolve();
   }
+
+  private updateFromResponse(model: ModelResponse): void {
+    // Update the readonly fields with the latest information.
+    (this.modelId as Model['modelId']) = extractModelId(model.name);
```

Comment:
Never a good idea to mutate readonly fields. Just store a `ModelResponse` as an attribute an expose the individual fields via getters. This method can hopefully just update the `ModelResponse` value.

```
private model: ModelResponse;

get modelId(): string {
  return this.model.modelId;
}
```

Reply:
done